### PR TITLE
[Doc] [NextJS] Error routes

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description
Covers topics:
* How does the Next.js 404 functionality work with JSS
* How does the Next.js error page work with JSS

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [ ] I have read the Contributing guide.
- [ ] My code follows the code style of this project.
- [ ] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change and it requires an update to the documentation.
- [ ] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

File: docs/data/routes/docs/nextjs/page-routing/error-routes/en.md [hunk 1]
```diff
@@ -3,4 +3,22 @@ name: error-routes
 routeTemplate: ./data/component-templates/article.yml
 title: Error Routes
 ---
-# Error Routes
\ No newline at end of file
+# Error Routes
```
Comment (ambrauer) [inline docs/data/routes/docs/nextjs/page-routing/error-routes/en.md:6]: Should we make this "Error Pages" instead (follows our names below and Next.js naming)? (here and in our nav)

**Review by ambrauer — APPROVED** [review]
Tech review looks good.

**Review by anastasiya29 — COMMENTED** [review]
A common question we get asked is, how to handle other response codes, like Not Authorized. There is some content in [Guides](https://github.com/Sitecore/jss/blob/guides/docs/data/routes/guides/error-handling-debugging/http-status-codes/en.md) on this; can any of that be reused here?

**Comment by illiakovalenko** [comment]
@anastasiya29 Nextjs 500 page handles all errors except of 404, I mentioned it in doc. Should I add something else?
https://github.com/Sitecore/jss/pull/611/files#diff-fb28a0ed6dea90ee06fd51c3366423ed1c871029e14e49d6a5a011376ccc51b3R22

File: docs/data/routes/docs/nextjs/page-routing/error-routes/en.md (hunk 1 above)
Comment (illiakovalenko) [reply docs/data/routes/docs/nextjs/page-routing/error-routes/en.md]: Agree, changed to "Error Pages"

**Comment by illiakovalenko** [comment]
@anastasiya29 Added paragraph how to render custom error code

**Review by ambrauer — APPROVED** [review]

**Pull request merged**